File: in_Java/0035 Search Insert Position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int[] res = new int[nums.length+1];
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == target) return i;
            res[i] = nums[i];
        }
        res[res.length-1] = target;
        Arrays.sort(res);
        for(int i = 0; i < res.length; i++){
            if(res[i] == target) return i;
        }
        return -1;
    }
}

File: in_Python/0035 Search Insert Position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        i,j = 0, len(nums)-1
        if target < nums[0]:
            return 0
        if target > nums[-1]:
            return j+1
        while i < j:
            m = i + (j-i)/2
            if nums[m] == target:
                return m
            elif nums[m] < target:
                i = m + 1
            else:
                j = m
        return i
        